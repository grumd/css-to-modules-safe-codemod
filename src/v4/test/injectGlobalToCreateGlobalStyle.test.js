'use strict';

const pluginTester = require('babel-plugin-tester');
const injectGlobalToCreateGlobalStyle = require('../injectGlobalToCreateGlobalStyle');
const prettier = require('prettier');

const tests = [
  {
    title: 'ES Import case',
    code: `
    import { injectGlobal } from 'styled-components';
    injectGlobal\`
      html { color: red; }
    \`;
    `,
  },
  {
    title: 'CommonJS, Non-destructured',
    code: `
    const styled = require('styled-components');
    styled.injectGlobal\`
      html { color: red; }
    \`;
    `,
  },
  {
    title: 'CommonJS, Destructured',
    code: `
    const { injectGlobal } = require('styled-components');

    injectGlobal\`
      html { color: red; }
    \`;
    `,
  },
];

pluginTester({
  pluginName: 'styled-components-v3-to-v4-global-style',
  plugin: injectGlobalToCreateGlobalStyle,
  snapshot: true,
  formatResult: output =>
    prettier.format(output, {
      semi: true,
      singleQuote: true,
      parser: 'babylon',
    }),
  tests: [...tests],
});
