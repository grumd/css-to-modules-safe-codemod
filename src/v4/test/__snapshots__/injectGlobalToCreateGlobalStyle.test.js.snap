// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled-components-v3-to-v4-global-style CommonJS, Destructured: CommonJS, Destructured 1`] = `
"
const { injectGlobal } = require('styled-components');

injectGlobal\`
  html { color: red; }
\`;

      ↓ ↓ ↓ ↓ ↓ ↓

const { createGlobalStyle } = require('styled-components');

const GlobalStyle = createGlobalStyle\`
  html { color: red; }
\`;
throw new Error(
  'An injectGlobal usage was converted to createGlobalStyles via codemod but needs to be hooked up. See https://www.styled-components.com/docs/api#createglobalstyle for instructions.'
);

"
`;

exports[`styled-components-v3-to-v4-global-style CommonJS, Non-destructured: CommonJS, Non-destructured 1`] = `
"
const styled = require('styled-components');
styled.injectGlobal\`
  html { color: red; }
\`;

      ↓ ↓ ↓ ↓ ↓ ↓

const styled = require('styled-components');
const GlobalStyle = styled.createGlobalStyle\`
  html {
    color: red;
  }
\`;
throw new Error(
  'An injectGlobal usage was converted to createGlobalStyles via codemod but needs to be hooked up. See https://www.styled-components.com/docs/api#createglobalstyle for instructions.'
);

"
`;

exports[`styled-components-v3-to-v4-global-style ES Import case: ES Import case 1`] = `
"
import { injectGlobal } from 'styled-components';
injectGlobal\`
  html { color: red; }
\`;

      ↓ ↓ ↓ ↓ ↓ ↓

import { createGlobalStyle } from 'styled-components';
const GlobalStyle = createGlobalStyle\`
  html { color: red; }
\`;
throw new Error(
  'An injectGlobal usage was converted to createGlobalStyles via codemod but needs to be hooked up. See https://www.styled-components.com/docs/api#createglobalstyle for instructions.'
);

"
`;
